### PR TITLE
[ENH] Add coverage

### PR DESCRIPTION
**Pull request by arnaudbore**
Combine with coverage jenkins plugins

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: Jenkinsfile [hunk 1]
```diff
@@ -24,13 +24,14 @@ pipeline {
...
                     sh '''
+                        pip3 install pytest-cov pytest-html
                         pip3 install wheel==0.38.*
                         pip3 install numpy==1.23.*
                         pip3 install packaging==23.*
                         pip3 install -e .
                         export MPLBACKEND="agg"
                         export OPENBLAS_NUM_THREADS=1
-                        pytest -v
+                        pytest --cov-report=xml --cov-report term-missing:skip-covered --cov=scilpy
```
Comment (AlexVCaron) [inline Jenkinsfile:34]: Remove --cov-report=xml, pytest.ini does the job already. Also, in my opinion, we want to have coverage for the scripts, right ? If so, remove, --cov=scilpy here, .coveragerc will take charge of setting this.

File: pytest.ini [hunk 2]
```diff
@@ -21,4 +21,18 @@ filterwarnings =
     once:::statsmodels
     once:::dmri-commit
     once:::cvxpy
-    once:::dmri-amico
\ No newline at end of file
+    once:::dmri-amico
+    
+required_plugins =
+    pytest-console-scripts
+    pytest-mock
```
Comment (AlexVCaron) [inline pytest.ini:28]: Completely missed this. Add `pytest-html` and `pytest-cov` here.

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
There is some little things I would change, but let's wait after the tests passes to do so and validate a first Coverage + Unit Tests report in Jenkins.

Also, don't forget to add pytest-html and pytest-cov in the requirements.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Pull request merged**